multi-connector/src/main/scala/it/agilelab/darwin/connector/multi/MultiConnector.scala

```diff
@@ -0,0 +1,324 @@
+package it.agilelab.darwin.connector.multi
+
+import it.agilelab.darwin.common.Connector
+import it.agilelab.darwin.common.compat.RightBiasedEither
+import it.agilelab.darwin.manager.SchemaPayloadPair
+import it.agilelab.darwin.manager.exception.DarwinException
+import it.agilelab.darwin.manager.util.{ AvroSingleObjectEncodingUtils, ConfluentSingleObjectEncoding }
+import org.apache.avro.Schema
+
+import java.io.{ ByteArrayInputStream, InputStream, OutputStream, SequenceInputStream }
+import java.nio.{ ByteBuffer, ByteOrder }
+
+class MultiConnector(
+  private[multi] val registrar: Connector,
+  private[multi] val confluentConnector: Option[Connector],
+  private[multi] val singleObjectEncodingConnectors: List[Connector]
+) extends Connector {
+
+  override def createTable(): Unit = registrar.createTable()
+
+  /**
+    * Returns whether or not the configured table exists
+    */
+  override def tableExists(): Boolean = registrar.tableExists()
+
+  /**
+    * @return a message indicating the user what he/she should do to create the table him/herself
+    */
+  override def tableCreationHint(): String = registrar.tableCreationHint()
+
+  /**
+    * Loads all schemas found on the storage.
+    * This method can be invoked multiple times: to initialize the initial values or to update the existing ones with
+    * the new data found on the storage.
+    *
+    * @return a sequence of all the pairs (ID, schema) found on the storage
+    */
+  override def fullLoad(): Seq[(Long, Schema)] =
+    (confluentConnector.toSeq.flatMap(_.fullLoad()) ++ singleObjectEncodingConnectors.flatMap(_.fullLoad())).distinct
+
+  /**
+    * Inserts all the schema passed as parameters in the storage.
+    * This method is called when new schemas should be registered in the storage (the test if a schema is already in
+    * the storage should be performed before the invocation of this method, e.g. by checking them against the
+    * pre-loaded cache).
+    *
+    * @param schemas a sequence of pairs (ID, schema) Schema entities to insert in the storage.
+    */
+  override def insert(schemas: Seq[(Long, Schema)]): Unit = registrar.insert(schemas)
+
+  /**
+    * Retrieves a single schema using its ID from the storage.
+    *
+    * @param id the ID of the schema
+    * @return an option that is empty if no schema was found for the ID or defined if a schema was found
+    */
+  override def findSchema(id: Long): Option[Schema] =
+    headOfIterator(
+      (confluentConnector.iterator ++ singleObjectEncodingConnectors.iterator)
```

Comment:
Maybe it is interesting to add the possibility to have the same function that do in parallel the lookup?
Also it has the "problem" that doesn't take in consideration the fact that for the same id more than one sub-connector can return a different result, I will also create the strict version that wait and make sure that the id isn't present in the others sub-connectors

Reply:
I documented that the connectors are looked up in order, but I'll add that confluent will have precedence (because reasons...), if an id is found twice, no error will be thrown but the first will be returned. It makes quite sense for schema fingerprints, less for incremental ids "confluent-style", but I do not think there is actually a "good way" of doing this. Let's give this feature a shot and then adjust this in next iterations